src/main/java/org/openstreetmap/atlas/checks/validation/points/AddressPointMatchCheck.java

```diff
@@ -0,0 +1,136 @@
+    {
+        // Object is an instance of Point
+        return object instanceof Point
+                // And has a street number specified
+                && object.getTag(ADDRESS_STREET_NUMBER_KEY).isPresent()
+                // And if the street name key has a value of null
+                && (!object.getTag(POINT_STREET_NAME_KEY).isPresent()
+                        // Or if the street name key is not present
+                        || !object.getTags().containsKey(POINT_STREET_NAME_KEY));
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Point point = (Point) object;
+        // Get a bounding box around the Point of interest
+        final Rectangle box = point.getLocation().boxAround(boundsSize);
+
+        // Get all Points inside the bounding box
+        final Iterable<Point> interiorPoints = point.getAtlas().pointsWithin(box);
+        // Convert the Iterable into a Stream for filtering, remove Points that have null as their
+        // street name as they cannot be candidates for a street for the Point of interest
+        final Set<Point> points = StreamSupport.stream(interiorPoints.spliterator(), false)
+                .filter(interiorPoint -> interiorPoint.getTag(POINT_STREET_NAME_KEY).isPresent())
+                .collect(Collectors.toSet());
+
+        // Get all Edges that intersect or are contained within the bounding box
+        final Iterable<Edge> interiorEdges = point.getAtlas().edgesIntersecting(box);
+        // Convert the Iterable into a Stream for filtering, remove Edge that have null as their
+        // street name as they cannot be candidates for a street for the Point of interest
+        final Set<Edge> edges = StreamSupport.stream(interiorEdges.spliterator(), false)
+                .filter(interiorEdge -> interiorEdge.getTag(EDGE_STREET_NAME_KEY).isPresent())
+                .collect(Collectors.toSet());
+
+        final Set<String> streetNameList = new HashSet<>();
+
+        // If there are no Points or Edges in the bounding box
+        if (points.isEmpty() && edges.isEmpty())
+        {
+            // Flag Point with instruction indicating that there are are no suggestions
+            return Optional.of(this.createFlag(point,
+                    this.getLocalizedInstruction(2, point.getOsmIdentifier())));
+        }
+        // If there are Points in the bounding box
+        else if (!points.isEmpty())
+        {
+            // Add all interior Point street names to the list of candidate street names
+            points.forEach(interiorPoint -> streetNameList
+                    .add(interiorPoint.getTags().get(POINT_STREET_NAME_KEY)));
```

Comment:
Street names could be collected by `points.forEach(point -> point.tag(POINT_STREET_NAME_KEY)).collect(Collectors.toSet())`.